lib/iris/cube.py

```diff
@@ -3052,7 +3052,7 @@ def _deepcopy(self, memo, data=None):
                 new_cube_data = self.data.copy()
         else:
             # Use the provided data (without copying it).
-            if not isinstance(data, biggus.Array):
+            if not isinstance(data, da.Array):
```

Comment:
Yes I believe the two are equivalent.  I have not tested the is_lazy_data approach, but I can't see a good reason not to use it.